core/protocols/aws-json-protocol/src/main/java/software/amazon/awssdk/protocols/json/internal/unmarshall/AwsJsonProtocolErrorUnmarshaller.java

```diff
@@ -87,6 +91,20 @@ private AwsServiceException unmarshall(SdkHttpFullResponse response, ExecutionAt
         return exception.build();
     }
 
+    private String getEffectiveErrorCode(SdkHttpFullResponse response, String errorCode) {
```

Comment:
Can we do this.hasAwsQueryCompatible at the beginning , can we break this function into smaller functions just as commented for V1 ?